Alignment/RPDataFormats/plugins/TotemRPIncludeAlignments.cc

```diff
@@ -0,0 +1,270 @@
+/****************************************************************************
+*
+* This is a part of TOTEM offline software.
+* Authors: 
+*  Jan Kaspar (jan.kaspar@gmail.com) 
+*
+****************************************************************************/
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/SourceFactory.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/Framework/interface/EventSetupRecordIntervalFinder.h"
+
+#include "Alignment/RPDataFormats/interface/RPAlignmentCorrectionsSequence.h"
+#include "CondFormats/AlignmentRecord/interface/RPMeasuredAlignmentRecord.h"
+#include "CondFormats/AlignmentRecord/interface/RPRealAlignmentRecord.h"
+#include "CondFormats/AlignmentRecord/interface/RPMisalignedAlignmentRecord.h"
+
+#include <vector>
+#include <string>
+#include <map>
+#include <set>
+
+
+/**
+ * \ingroup TotemRPGeometry
+ * \brief A class adding (mis)alignments to geometry (both real and misaligned).
+ *
+ * See schema of \ref TotemRPGeometry "TOTEM RP geometry classes"
+ **/
+class  TotemRPIncludeAlignments : public edm::ESProducer, public edm::EventSetupRecordIntervalFinder
+{
+  public:
+    TotemRPIncludeAlignments(const edm::ParameterSet &p);
+    virtual ~TotemRPIncludeAlignments(); 
+
+    std::auto_ptr<RPAlignmentCorrections> produceMeasured(const RPMeasuredAlignmentRecord &);
+    std::auto_ptr<RPAlignmentCorrections> produceReal(const RPRealAlignmentRecord &);
+    std::auto_ptr<RPAlignmentCorrections> produceMisaligned(const RPMisalignedAlignmentRecord &);
```

Comment:
A recent framework change now requires all the return values to be `std::unique_ptr<>`. Sorry you got caught in the middle of the transition.